src/crawlee/crawlers/_basic/_basic_crawler.py

```diff
@@ -1226,3 +1250,18 @@ def _raise_for_session_blocked_status_code(self, session: Session | None, status
             ignore_http_error_status_codes=self._ignore_http_error_status_codes,
         ):
             raise SessionError(f'Assuming the session is blocked based on HTTP status code {status_code}')
+
+    def _raise_request_collision(self, request: Request, session: Session | None) -> None:
```

Comment:
I'd prefer to call this `check_request_collision` or something like that. Now the name sounds like it's always going to raise an error, which is not the case.